src/check_nesting.cpp

```diff
@@ -0,0 +1,53 @@
+#include "sass.hpp"
+#include <vector>
+
+#include "check_nesting.hpp"
+#include "context.hpp"
+
+namespace Sass {
+
+  CheckNesting::CheckNesting(Context& ctx)
+  : ctx(ctx),
+    block_stack(std::vector<Block*>())
+  { }
+
+  Statement* CheckNesting::operator()(Block* b)
+  {
+    Block* bb = SASS_MEMORY_NEW(ctx.mem, Block, b->pstate(), b->length(), b->is_root());
```

Comment:
Is it really necessary to make another copy? I figure this should not be needed since we do not really alter anything here? Creating too many clones is the main reason why libsass is slower than it could be (I already removed some cloning in eval some time ago, which lead to the big performance gain). The other reason we are slower than needed is that we visit too many visitors, but I guess there is not much too avoid that ...